### PR TITLE
Dont overwrite dotnet env vars when transforming projects into containers

### LINKED ISSUE #1476: How to overwrite ASPNETCORE_ENVIRONMENT when running in Docker?
I have a bunch of appsetting json files for different environments, like for instance appsettings.local.json. I'm loading the environment specific file like this:
`config.AddJsonFile($"appsettings.{hostingContext.HostingEnvironment.EnvironmentName}.json", optional: false, reloadOnChange: true);`

In my yaml file I set 
```
name: MyServices
services:
  - name: SomeApi
    project: SomeApi/SomeApi.csproj
    env:
      - name: ASPNETCORE_ENVIRONMENT
        value: "local"
```

When running `tye run` this is fine, but using `--docker` seems to overwrite the env variable with "Development". I can see this in the logs as wel, where I can see `-e "ASPNETCORE_ENVIRONMENT=Development"` being passed on container initialization. 

How can I set ASPNETCORE_ENVIRONMENT to the value I want?

### PR DESCRIPTION
**Pull request by josdeweger**
Fixes #1476

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=1480) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=1480) before we can accept your contribution.<br/><br/>:x: josdeweger [sign now](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=1480)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/tye?pullRequest=1480) it.</sub>

**Comment by davidfowl** [comment]
@josdeweger The whitespace police came for your PR 😄

File: src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs [hunk 1]
```diff
@@ -17,6 +17,18 @@ public class TransformProjectsIntoContainers : IApplicationProcessor
         private readonly ILogger _logger;
         private Lazy<TempDirectory> _certificateDirectory;
 
+        private Dictionary<string, string> _defaultDotnetEnvVars = new Dictionary<string, string>
```
Comment (davidfowl) [inline src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs:20]:
```suggestion
        private static readonly Dictionary<string, string> _defaultDotnetEnvVars = new Dictionary<string, string>
```

File: src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs [hunk 2]
```diff
@@ -17,6 +17,18 @@ public class TransformProjectsIntoContainers : IApplicationProcessor
...
+        {
+            { "DOTNET_ENVIRONMENT", "Development" },
+            { "DOTNET_LOGGING__CONSOLE__DISABLECOLORS", "true" }
+        };
+
+        private Dictionary<string, string> _defaultAspnetEnvVars = new Dictionary<string, string>
```
Comment (davidfowl) [inline src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs:26]:
```suggestion
        private static readonly Dictionary<string, string> _defaultAspnetEnvVars = new Dictionary<string, string>
```

**Comment by davidfowl** [comment]
Thanks for the fix!

**Comment by josdeweger** [comment]
> Thanks for the fix!

Thanks for merging @davidfowl !

**Comment by davidfowl** [comment]
Thanks for the change!

**Pull request merged**